pir/cpp/encoder.h

```diff
@@ -0,0 +1,174 @@
+//
```

Comment:
As we discussed, let's keep the encoders simple for now, since we'll probably not end up using them at all in the near future.